src/main/scala/org/allenai/scholar/eval/CreateGoldData.scala

```diff
@@ -65,34 +66,26 @@ object DblpFromPipeline extends App with CreateGoldData {
 
   import MetadataParser._
 
-  val rand = new Random()
-  var size = 1000.0
-  var count = 1.0
-  val clusters = new ListBuffer[Cluster]
   val keepId: String => Boolean = {
     val idFile = new File("src/main/resources/raw/dblp/ids.txt")
-    if (idFile.exists) {
-      // Use ID file if it exists
-      val ids = Resource.using(Source.fromFile(idFile)) { src => src.getLines.toSet }
-      s: String => ids(s)
-    } else {
-      // Resevoir sample 1000 papers
-      s: String => (rand.nextDouble < size / count)
-    }
+    val ids = Resource.using(Source.fromFile(idFile)) { src => src.getLines().toSet }
+    s: String => ids(s)
   }
-  Resource.using(Source.fromFile(pipelineMetadataOutputFile)) {
-    src =>
-      for {
-        c @ Cluster(ids, acl, Some(dblp), grobid, scienceParse) <- src.getLines().map(parse) if keepId(ids.head)
-      } {
-        if (count <= size) {
-          clusters += c
-        } else {
-          clusters(rand.nextInt(clusters.size)) = c
-        }
-        count += 1
+
+  val allClusters = Resource.using(Files.list(Paths.get("/Users/dirkg/temp/FinalPaperClusters.fa33b8c61c12c42b3e240e2dc7bc1e44591619aa"))) { javaPaths =>
+    val paths = javaPaths.collect(Collectors.toList()).asScala.toIterator
+    paths.filter(p => p.getFileName.toString.startsWith("part-") && p.getFileName.toString.endsWith(".gz")).flatMap { path =>
+      println(path.toString + " ...")
+      Resource.using(Source.fromInputStream(new GZIPInputStream(Files.newInputStream(path)))) { source =>
+        source.getLines().map(parse).toList
       }
+    }
```

Comment:
Instead of reading from your custom pipeline step, this can read the output of FinalPaperClusters. It's not that much data. Takes about 30 seconds to finish.